metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/hook/UpdateIndicesHook.java

```diff
@@ -160,11 +175,13 @@ private void handleUpdateChangeEvent(@Nonnull final MetadataChangeLog event) {
     }
 
     // Step 1. For all aspects, attempt to update Search
-    updateSearchService(entitySpec.getName(), urn, aspectSpec, aspect,
-        event.hasSystemMetadata() ? event.getSystemMetadata().getRunId() : null);
+    updateSearchService(entitySpec.getName(), urn, aspectSpec, aspect, event.getSystemMetadata(), previousAspect);
 
     // Step 2. For all aspects, attempt to update Graph
-    if (_diffMode) {
+    SystemMetadata systemMetadata = event.getSystemMetadata();
+    if (_graphDiffMode && _graphService instanceof ElasticSearchGraphService
```

Comment:
This condition is not easy to follow

Reply:
It's essentially two flag checks with mandatory null checks. The class check is no longer necessary here thanks to an OSS contribution.